packages/code-combinators/test/Test/CodeCombinators/GenPat.hs

```diff
@@ -0,0 +1,106 @@
+module Test.CodeCombinators.GenPat where
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import qualified Language.Haskell.TH as TH
+import qualified Happy.Backend.CodeCombinators.Syntax as SnGen
+import qualified Happy.Backend.CodeCombinators.Abstract()
+import Happy.Backend.CodeCombinators
+import Test.CodeCombinators.Common
+import Data.List
+
+
+genLitP :: MonadGen m => m TH.Pat
+genLitP = do
+  x <- Gen.choice $
+         [
+            do
+            c <- Gen.latin1
+            return $ litP $ TH.CharL c
+          , do
+            s <- Gen.list (Range.linear 0 20) Gen.latin1
+            return $ litP $ TH.StringL $ delete '\n' s
+          , do
+            x <- Gen.int $ Range.linear minBound maxBound
+            return $ litP $ TH.IntegerL $ fromIntegral x
```

Comment:
```suggestion
          , (litP . TH.IntegerL) <$> Gen.integral (Range.constantBounded @Int)
```
(Did not try to see if it will work, but something like this. I’m also not sure if you really want the whole `Int` range or something smaller would suffice too.)